cli/src/commands/router/commands/download.ts

```diff
@@ -0,0 +1,132 @@
+import fs from 'node:fs';
+import { Octokit } from 'octokit';
+import { Command, program } from 'commander';
+import pc from 'picocolors';
+import cliProgress from 'cli-progress';
+import decompress from 'decompress';
+import { join } from 'pathe';
+import { BaseCommandOptions } from '../../../core/types/types.js';
+
+export default function (_: BaseCommandOptions) {
+  const command = new Command('download');
+  command.description('Downloads the latest router binary for the detected machine platform and architecture.');
+  command.option('-o, --out [string]', 'Destination directory for the downloaded router binary.');
+  command.action(async (options) => {
+    const path = join(process.cwd(), options.out ?? './router');
+    const fullPath = path + '/router';
+    if (fs.existsSync(fullPath)) {
+      console.log(pc.red(`${path}/router already exists`));
```

Comment:
I'd assume it should exit 1 when the download operation could not be run successfully

Reply:
Would you like me to change this to an error?